### ISSUE TITLE
trafoValue ignores names

### ISSUE THREAD

**Issue by danielhorn** [berndbischl/ParamHelpers#102]
I had this bug in one of my applications:

```library(ParamHelpers)
ps = makeParamSet(
  makeNumericParam("cost", lower = -15L, upper = 15L, trafo = function(x) 2^x),
  makeNumericParam("gamma", lower = -15L, upper = 15L, trafo = function(x) 2^x),
  makeNumericParam("wcw.weight", lower = -7L, upper = 7L, trafo = function(x) 2^x)
)

set.seed(2)
par = sampleValue(ps)
names(par)[3] = "bla"
trafoValue(ps, par[c(1, 2, 3)])
$cost
[1] 0.001426275

$gamma
[1] 67.23875

$wcw.weight
[1] 2.037173
trafoValue(ps, par[c(2, 1, 3)])
$cost
[1] 67.23875

$gamma
[1] 0.001426275

$wcw.weight
[1] 2.037173
```
Please note: The transformed values of cost and gamma are switched and the name "bla" of the third parameter has been reset to wcw.weight.

How to fix this? Check names and throw an error, if names of par and names in the param.set do not match? What if the order is incorrect? Reorder automatically or throw an error?

**Comment by berndbischl** [comment]
well, you are forcefully changing names here. 
why do you think this should work? you are changing the structure of the object behind the functions back....

**Comment by berndbischl** [comment]
probably trafoValues should CHECK whether the names in the list are the same as the names of the params in ps......

**Comment by danielhorn** [comment]
I don't expect this to work. I expect this to throw an error and to not do "something" with a wrong object.

There should also be an check for forbidden regions and requirements? So, we probably want to call isFeasible here.

**Comment by jakob-r** [comment]
>you are changing the structure of the object behind the functions back....

Not so true, par is just a list. But as it's stated in the manpage

>For a parameter set this must be a list in the correct order.

So the order is all that matters. I remember that I wrote something similar for isFeasible in my branch with PR #73. Maybe we can generalize and prevent copy pasta:
https://github.com/berndbischl/ParamHelpers/blob/fix_40_isFeasible/R/isFeasible.R#L59-L83

So we should

1. test if named
2. if named check if names match and apply trafo based on names
3. if unnamed check that length matches and apply trafo in order

**Comment by berndbischl** [comment]
we basically did 1-3 what @jakob-r said

**Comment by berndbischl** [comment]
fixed in PR #122
